src/borg/repository.py

```diff
@@ -262,14 +262,14 @@ def open(self, path, exclusive, lock_wait=None, lock=True):
         if 'repository' not in self.config.sections() or self.config.getint('repository', 'version') != 1:
             self.close()
             raise self.InvalidRepository(path)
-        self.max_segment_size = self.config.getint('repository', 'max_segment_size')
+        self.peripheral_segment_size = self.config.getint('repository', 'max_segment_size')
```

Comment:
here, you read 'max_segment_size' from the config - which works for current configs, but won't work for configs created by line 171, which would have 'peripheral_segment_size'.

So, this has misc. problems:
- it does not work
- names are less consistent now than before
- "peripheral_segment_size" is not really a better name
- still uses DEFAULT_MAX_SEGMENT_SIZE
- you just changed names, not behaviour.

I'ld suggest to first make a solid plan about what to do about this (if anything) in the ticket, agree on it and then work on the code.